connect/connect.go

```diff
@@ -0,0 +1,292 @@
+package connect
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/kyma-incubator/hydroform/connect/types"
+	"io/ioutil"
+	"log"
+	"net/http"
+)
+
+func (c *KymaConnector) Connect(configurationUrl string) error {
+
+	_, err := c.populateCsrInfo(configurationUrl)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	_, err = c.populateClientCert()
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	if c.SecureClient == nil {
+		err = c.populateClient()
+		if err != nil {
+			return fmt.Errorf(err.Error())
+		}
+	}
+
+	_, err = c.populateInfo()
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	err = c.StorageInterface.WriteClientCert(c.Ca)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	return err
+}
+
+func (c *KymaConnector) RegisterService(serviceDescription Service) (serviceId string, err error) {
+	jsonBytes, err := json.Marshal(serviceDescription)
+	if err != nil {
+		return "", fmt.Errorf(err.Error())
+	}
+
+	if c.CsrInfo == nil || c.CsrInfo.API.MetadataUrl == "" {
+		return "", fmt.Errorf(err.Error())
+	}
+
+	resp, err := c.SecureClient.Post(c.CsrInfo.API.MetadataUrl, "application/json", bytes.NewBuffer(jsonBytes))
+	if err != nil {
+		return "", fmt.Errorf(err.Error())
+	}
+	defer resp.Body.Close()
+
+	bodyBytes, _ := ioutil.ReadAll(resp.Body)
+	bodyString := string(bodyBytes)
+
+	if err != nil {
```

Comment:
What error is being handled here? The `ReadAll` error is being ignored.
General pointers for errors in Go:
- Errors should be treated as close as possible to where they happened, ideally right away.
- Try not to reuse a variable for errors since it masks bugs in error handling.